services/spar/src/Spar/Scim/User.hs

```diff
@@ -315,29 +283,36 @@ updateValidScimUser tokinfo uidText newScimUser = do
     -- TODO: how do we get this safe w.r.t. race conditions / crashes?
 
     -- construct old and new user values with metadata.
-    uid :: UserId <- parseUid uidText
-    oldScimStoredUser :: Scim.StoredUser ScimUserExtra
-      <- let err = throwError $ Scim.notFound "user" uidText
-         in maybe err pure =<< Scim.get tokinfo uidText
+    oldScimStoredUser :: Scim.StoredUser SparTag
+      <- Scim.getUser tokinfo uid
 
-    let userRef = newScimUser ^. vsuSAMLUserRef
-    assertUserRefUnused uid userRef
+    assertUserRefNotUsedElsewhere (newScimUser ^. vsuSAMLUserRef) uid
+    assertHandleNotUsedElsewhere (newScimUser ^. vsuHandle) uid
 
     if Scim.value (Scim.thing oldScimStoredUser) == (newScimUser ^. vsuUser)
       then pure oldScimStoredUser
       else do
-        newScimStoredUser :: Scim.StoredUser ScimUserExtra
+        newScimStoredUser :: Scim.StoredUser SparTag
           <- lift $ updScimStoredUser (newScimUser ^. vsuUser) oldScimStoredUser
 
-        -- update 'SAML.UserRef'
-        let uref = newScimUser ^. vsuSAMLUserRef
-        lift . wrapMonadClient $ Data.insertSAMLUser uref uid  -- on spar
-        bindok <- lift $ Intra.Brig.bindBrigUser uid uref  -- on brig
+        -- update 'SAML.UserRef' on spar
+        let newuref = newScimUser ^. vsuSAMLUserRef
+        lift . wrapMonadClient $ Data.insertSAMLUser newuref uid
+        midp :: Maybe IdP <- maybe (pure Nothing) (lift . wrapMonadClient . Data.getIdPConfig) stiIdP
+        forM_ midp $ \idp -> do
+          let eid = Scim.externalId . Scim.value . Scim.thing $ oldScimStoredUser
+          olduref <- mkUserRef idp eid
+          lift . wrapMonadClient $ Data.deleteSAMLUser olduref
```

Comment:
If `olduref == newuref`, we're going to insert it and immediately delete it, which looks like a bug. If I'm not mistaken and it's indeed a bug, let's add a test for it.

Reply:
yes, seems to me it's a bug.  i'll add a test case, thanks!

Reply:
hum, the enhanced test from 42edf8b already passes without the subsequent fix.  both prod code and test seem fine to me *now*, but it puzzles me that the old prod code doen't make the test fail.  i will take another look tomorrow.

Reply:
the test was still wrong.  with 0e04f99, the bug you found can be reproduced, and the tests pass with the fix.  all good now, i think!